docs/DeveloperGuide.md

```diff
@@ -6,32 +6,215 @@
+2. The handler passes this to a corresponding Parser object for that command, which will split it into its necessary parts
+3. The Parser will create a corresponding Command object for that command, which will perform its necessary actions
+4. The Command object will return successfully to the Command Handler, and the output will be given to the user (output for the user is stored in a `Response` class)
+
+The command to add a project is one example of this structure:
+
+#### Add a Project
+![image info](./UmlDiagrams/addProjectNew.png)
+
+**Step1.** After recieving the necessary method call, the `CommandHandler` class will self-call executeCommand with both the list of projects as well as a constructor call to AddProjectCommandParser (and a call to the parse method of AddProjectCommandParser) as its parameters
+
+**Step2.** The AddProjectCommandParser object instance is created
+
+**Step3.** The parse method of AddProjectCommandParser is called with `commands` (the user input) as its parameter.
+
+**Step4.** AddProjectCommandParser will self call its `getProjectTitle` method. This will split the user's input and return the correct name for the project
+
+**Step5.** After returning this projectTitle to itself, `AddProjectCommandParser` will call the constructor for a new `AddProjectCommand` object
+
+**Step6.** This `AddProjectCommand` object will use the newly gathered project name to call the `addProjct` command of the `ProjectList` class.
+
+**Step7.** The status of this command completing is returned to `CommandHandler`, and the output is sent to the user.
+
+Deleting a project will follow nearly the exact same structure; the only difference is that the corresponding Parser and Command objects will be used (`DeleteProjectCommandParser` and `DeleteProjectCommand`)
+
+####Delete a Project
+![image info](./UmlDiagrams/deleteProjectNew.png)
+*The steps are omitted here as they are the exact same as adding a project, only with the differing class names previously mentioned.
+
+#### Print Project List
+![image info](./UmlDiagrams/printProject.png)
+
+**Step1.** When `CommandHandler` receives a user input starting with string "listproject" or "listprojects", it will call `listProjects`. This method ensure that `projectList` has been initialized before the next step.
+
+**Step2.** Once the first step is done, `listProjects` will call the `printProject` method from `ProjectList` with no parameter.
+
+**Step3.** `printList` will first check if the size of the private ArrayList named `projectList` is 0, which means that the arraylist is empty. If yes, it will print "You have not added any projects yet!" and return.
+
+**Step4.** It will then iterate over `projectList`, which type is `Project`. It will call `getTitle` and `getDeadline` from each array element in`the `Project` class.
+
+**Step5.** the general information of a project, i.e., its title and deadline, will be displayed.
+
+#### Todo feature
+![image info](./UmlDiagrams/Todo.png)  
```

Comment:
1. Perhaps the return arrow can touch the activation bar of the self-invocation to make it less ambiguous
![image](https://user-images.githubusercontent.com/69310031/160963650-2d7dd078-6511-48cc-8124-4191872b4c5f.png)

2. targetProject returned can possibly add the type and make sure that the arrow is end of the activation bar
![image](https://user-images.githubusercontent.com/69310031/160963789-d18902e7-9334-44ad-9d1a-93ab7db1e464.png)

3. Should there be a return arrow from the self-invocation activation bar?
![image](https://user-images.githubusercontent.com/69310031/160963844-b56ab941-734f-4cef-95fc-5cca92a90d56.png)

4. Possible to remove the italics from close bracket of the condition in the opt box (Should there be a return arow as well?)
![image](https://user-images.githubusercontent.com/69310031/160963923-a5f68089-23ec-4d1b-a4dd-a6a2da60a4df.png)